custom_auth/tests.py

```diff
@@ -33,67 +77,240 @@ def delete(self, *args, **kwargs):
-        user.set_password(password or self.password)
-        user.save()
-        return user
-
     def test_not_authenticated(self):
         # Should only be able to create a new user without logging in
         user = self.create_user()
 
-        response = self.get(views.UserView, '/user/', {'id': user.id})
+        response = self.get(views.UserView, {'id': user.id})
         self.assertEquals(response.status_code, 401, msg=response.content)
 
-        response = self.put(views.UserView, '/user/', {'id': user.id, 'email': user.email})
+        response = self.put(views.UserView, {'id': user.id, 'email': user.email})
         self.assertEquals(response.status_code, 401, msg=response.content)
 
-        response = self.delete(views.UserView, '/user/')
+        response = self.delete(views.UserView)
         self.assertEquals(response.status_code, 401, msg=response.content)
 
     def test_create_user(self):
-        response = self.post(views.UserView, '/user/', {'email': self.email, 'password': self.password})
+        response = self.post(views.UserView, {'email': self.email, 'password': self.password})
         self.assertEquals(response.status_code, 201, msg=response.content)
         self.assertEquals(response.data['user']['email'], self.email, msg=response.content)
         self.assertIn('refresh', response.data, msg=response.content)
         self.assertIn('access', response.data, msg=response.content)
+
+    def test_update_user(self):
+        user = self.create_user()
+        self.force_login(user)
+
+        first_name = "Test"
+        last_name = "Testing"
+        data = {'email': user.email, 'first_name': first_name, 'last_name': last_name}
+        response = self.put(views.UserView, data)
+
+        self.assertEquals(response.status_code, 200, msg=response.content)
+        self.assertEquals(response.data['email'], self.email, msg=response.content)
+        self.assertEquals(response.data['first_name'], first_name, msg=response.content)
+        self.assertEquals(response.data['last_name'], last_name, msg=response.content)
+
+        user = User.objects.get(pk=user.pk)
+        self.assertEquals(user.first_name, first_name, msg=response.content)
+        self.assertEquals(user.last_name, last_name, msg=response.content)
+
+    def test_get_user(self):
+        user = self.create_user()
+        self.force_login(user)
+
+        response = self.get(views.UserView, {'id': user.id})
+        self.assertEquals(response.status_code, 200, msg=response.content)
+        self.assertEquals(response.data['email'], self.email, msg=response.content)
+
+    def test_get_user_by_email(self):
+        user = self.create_user()
+        self.force_login(user)
+
+        response = self.get(views.UserByEmailView, {'email': user.email})
+        self.assertEquals(response.status_code, 200, msg=response.content)
+        self.assertEquals(response.data['id'], user.id, msg=response.content)
+
+    def test_delete_user(self):
```

Comment:
Should we test that a user cannot delete another user?

Reply:
When deleting an user doesn't take an id, it uses the currently logged in user